### PR TITLE
Release that supports Julia 0.6

### PR DESCRIPTION
**Pull request by abhijithanilkumar**
* Use StaticArrays instead of FixedSizeArrays
* Fix Deprecation warnings
* Fix type issues

### CODE REVIEW

File: test/runtests.jl [hunk 1]
```diff
@@ -151,8 +151,8 @@ jagmesh_adj = jagmesh()
           for i in 1:size(a,1)
              p = Int32[]
              for e in collect(edges(g))
-                 if e[1] == i
-                     push!(p,e[2])
+                 if e.src == i
```
Comment (sbromberger) [inline test/runtests.jl]: This should be `src(e)` for LightGraphs - use the accessor function, not the internal field.

Same with L155 - `dst(e)`.

File: src/stress.jl [hunk 2]
```diff
@@ -50,35 +50,35 @@ Reference:
...
     abstolx::FT
 end
 
 
-function initialweights(D, T=eltype(D))
+function initialweights(D)
     map(D) do d
-        x = T(d^(-2.0))
-        isfinite(x) ? x : zero(T)
+        x = Float64(d^(-2.0))
```
Comment (SimonDanisch) [inline src/stress.jl]: why would you want to restrict it only to Float64?

Comment (abhijithanilkumar) [inline src/stress.jl]: When generalizing, in 0.6 it's creating a sparse matrix of type `{Any,Int64}`. In 0.5 it was of type `{T,Int64}`. So in 0.6 this is throwing an error while calling zeros() function later in `weightedlaplacian`. Also, isn't it safe to assume that the output of d^-2.0 is always floating point?

Comment (SimonDanisch) [inline src/stress.jl]: >Also, isn't it safe to assume that the output of d^-2.0 is always floating point?

Float64 is not the only float type in julia...

And Any as the matrix eltype shouldn't happen, so you're just working around a bug that must have happened earlier!

Comment (abhijithanilkumar) [inline src/stress.jl]: Not the eltype, map function is generating different outputs in Julia 0.5 and 0.6
In 0.5, it generates sparse matrix with Float elements : 
```
julia> function fx(a,t=eltype(a))
              map(a) do d
              x=t(d^(-2.0))
              isfinite(x)?x:zero(t)
              end
              end
fx (generic function with 2 methods)

julia> fx(a,Float32)
30×30 sparse matrix with 116 Float32 nonzero entries:
	[2 ,  1]  =  1.0
	[3 ,  1]  =  1.0
	[4 ,  1]  =  1.0
	[5 ,  1]  =  1.0
	[6 ,  1]  =  1.0
	[7 ,  1]  =  1.0
	[8 ,  1]  =  1.0
```

In 0.6 it is generating matrix with Any elements : 
```
julia> function fx(a,t=eltype(a))
              map(a) do d
              x=t(d^(-2.0))
              isfinite(x)?x:zero(t)
              end
              end
fx (generic function with 2 methods)

julia> fx(a,Float32)
30×30 SparseMatrixCSC{Any,Int64} with 116 stored entries:
  [2 ,  1]  =  1.0
  [3 ,  1]  =  1.0
  [4 ,  1]  =  1.0
  [5 ,  1]  =  1.0
  [6 ,  1]  =  1.0
  [7 ,  1]  =  1.0
  [8 ,  1]  =  1.0
```
What do you suggest?

Comment (SimonDanisch) [inline src/stress.jl]: It must be `d^t(-2.0)` then...

Comment (abhijithanilkumar) [inline src/stress.jl]: Even `d^t(-2.0)` produces matrix with Any elements.

Comment (SimonDanisch) [inline src/stress.jl]: i can't reproduce this. Does a already have any as element type?

Comment (abhijithanilkumar) [inline src/stress.jl]: No. a is `SparseMatrixCSC{Int64,Int64}`, which is the output of `adjacency_matrix` function of LightGraphs.

Comment (SimonDanisch) [inline src/stress.jl]: Ah lol.... So eltype is actually Int? Don't you get a domain error then?

Comment (abhijithanilkumar) [inline src/stress.jl]: That was a bug. But it never came up because an argument was always passed to the function.

Comment (SimonDanisch) [inline src/stress.jl]: Okay, so the actual bug is to default to eltype then...
That means, you need to pass T down to initial weights.
And then write it as:
```Julia
 x = T(d) ^ T(-2.0)
```

Comment (abhijithanilkumar) [inline src/stress.jl]: Passing T to initialweights was the original setup. And doing `x = T(d) ^ T(-2.0)` still generates Any.

Comment (sbromberger) [inline src/stress.jl]: Sorry, just jumping in here - haven't read the entire comment thread but wanted to make sure you knew that `adjacency_matrix(::AbstractGraph, ::DataType)` was an option:
```
julia> g = Graph(3,3)
{3, 3} undirected simple Int64 graph

julia> adjacency_matrix(g, Float64)
3×3 SparseMatrixCSC{Float64,Int64} with 6 stored entries:
  [2, 1]  =  1.0
  [3, 1]  =  1.0
  [1, 2]  =  1.0
  [3, 2]  =  1.0
  [1, 3]  =  1.0
  [2, 3]  =  1.0
```

again, not sure if relevant or helpful, but there it is.

Comment (abhijithanilkumar) [inline src/stress.jl]: That still doesn't solve the issue.

Comment (abhijithanilkumar) [inline src/stress.jl]: Is there any possible workaround for this issue other than restricting to a particular Float type?

Comment (SimonDanisch) [inline src/stress.jl]: yes

Comment (abhijithanilkumar) [inline src/stress.jl]: Tried the above suggestions and still no luck :/

Comment (SimonDanisch) [inline src/stress.jl]: Sorry, I have absolutely no time to care about this package right now. 
If you can't figure it out, I guess that's then just how it is.

Comment (abhijithanilkumar) [inline src/stress.jl]: Thanks for your time :) I'll try to find a way.

Comment (sbromberger) [inline src/stress.jl]: I think I've figured out how to do this:
```
julia> function fx(a,t=eltype(a))::SparseMatrixCSC{t, Int64}
                     map(a) do d
                     x=t(d^(-2.0))
                     isfinite(x)?x:zero(t)
                     end
                     end
fx (generic function with 2 methods)

julia> g = Graph(3,3)
{3, 3} undirected simple Int64 graph

julia> fx(adjacency_matrix(g), Float32)
3×3 SparseMatrixCSC{Float32,Int64} with 6 stored entries:
  [2, 1]  =  1.0
  [3, 1]  =  1.0
  [1, 2]  =  1.0
  [3, 2]  =  1.0
  [1, 3]  =  1.0
  [2, 3]  =  1.0
```

Return type annotation.

Comment (abhijithanilkumar) [inline src/stress.jl]: Thank you for the suggestion! That fixes the issue.

**Pull request merged**